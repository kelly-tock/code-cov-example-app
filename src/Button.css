.Button {
  border-radius: 20px;
  color: black;
  padding: 6px 12px;
  border: 1px solid rgb(250, 128, 114);
  background-color: white;
  transition: border, background-color 0.3s ease-in-out;
}

.Button:hover {
  background-color: rgba(250, 128, 114, 0.6);
  border-color: rgba(250, 128, 114, 0.6);
}

.Button--primary {
  border: 1px solid rgb(134, 114, 250);
  background-color: white;
}

.Button--primary:hover {
  background-color: rgba(134, 114, 250, 0.6);
  border-color: rgba(134, 114, 250, 0.6);
}

.Button--secondary {
  border: 1px solid rgb(121, 121, 121);
  background-color: white;
}

.Button--secondary:hover {
  background-color: rgba(121, 121, 121, 0.6);
  border-color: rgba(121, 121, 121, 0.6);
}
