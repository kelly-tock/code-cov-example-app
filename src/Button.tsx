import React, { FunctionComponent } from "react";
import clsx from "clsx";
import "./Button.css";

interface ButtonProps {
  variant?: "primary" | "secondary";
  size?: "small" | "medium" | "large";
  startIcon?: React.ReactNode;
}

export const Button: FunctionComponent<ButtonProps> = ({
  variant,
  size = "medium",
  children,
  startIcon,
}) => {
  return (
    <button
      className={clsx("Button", {
        [`Button--${variant}`]: variant,
        [`Button--${size}`]: size,
      })}
    >
      {startIcon && <span className="Button--startIcon">{startIcon}</span>}
      {children}
    </button>
  );
};
