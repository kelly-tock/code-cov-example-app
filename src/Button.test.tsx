import React from "react";
import { render, screen } from "@testing-library/react";
import { Button } from "./Button";

test("matches default snap", () => {
  const { asFragment } = render(<Button>test default button</Button>);
  expect(asFragment).toMatchSnapshot();
});

test("renders a start icon", () => {
  const { asFragment } = render(
    <Button startIcon={<>&lozenge;</>}>test default button</Button>
  );
  expect(asFragment).toMatchSnapshot();
});

test("renders an end icon", () => {
  const { asFragment } = render(
    <Button endIcon={<>&loang;</>}>test default button</Button>
  );
  expect(asFragment).toMatchSnapshot();
});
