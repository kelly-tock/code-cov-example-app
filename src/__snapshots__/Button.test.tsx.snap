// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches default snap 1`] = `[Function]`;

exports[`renders a start icon 1`] = `[Function]`;

exports[`renders an end icon 1`] = `[Function]`;
